### ISSUE TITLE
Add OAuth2 authentication for Perl auth0

### ISSUE THREAD

**Issue by preaction** [cpan-testers/cpantesters-web#13]
We need some form of authentication before we can offer features like messaging and marking reports as invalid. The Perl NOC is moving away from Bitcard and towards auth0, which is an OAuth2 provider.

We need to add an OAuth2 consumer that allows users to log in using the auth0 provider. This should probably use Mojolicious::Plugin::OAuth2. Once we've set that up, we need to tell Robert what our OAuth2 callback URL is so they can enable us and give us our shared secret.

This does not include creating user accounts or linking them to PAUSE accounts. That will need to be handled in another ticket. This is just to get access to the auth0 instance so we can start thinking about how we want our authorization / user accounts to work.
